src/conversions/chrono.rs

```diff
@@ -0,0 +1,681 @@
+        let datetime = PyDateTime::new_with_fold(py, yy, mm, dd, h, m, s, ms, Some(tz), fold)
+            .expect("Failed to construct datetime");
+        datetime.into()
+    }
+}
+
+impl<Tz: TimeZone> IntoPy<PyObject> for DateTime<Tz> {
+    fn into_py(self, py: Python<'_>) -> PyObject {
+        ToPyObject::to_object(&self, py)
+    }
+}
+
+impl FromPyObject<'_> for DateTime<FixedOffset> {
+    fn extract(ob: &PyAny) -> PyResult<DateTime<FixedOffset>> {
+        let dt = <PyDateTime as PyTryFrom>::try_from(ob)?;
+        let ms = dt.get_fold() as u32 * 1_000_000 + dt.get_microsecond();
+        let h = dt.get_hour().into();
+        let m = dt.get_minute().into();
+        let s = dt.get_second().into();
+        let tz = if let Some(tzinfo) = dt.get_tzinfo() {
+            tzinfo.extract()?
+        } else {
+            return Err(PyTypeError::new_err("Not datetime.timezone.tzinfo"));
+        };
+        let dt = NaiveDateTime::new(
+            NaiveDate::from_ymd(dt.get_year(), dt.get_month().into(), dt.get_day().into()),
+            NaiveTime::from_hms_micro(h, m, s, ms),
+        );
+        Ok(DateTime::from_utc(dt, tz))
+    }
+}
+
+impl FromPyObject<'_> for DateTime<Utc> {
+    fn extract(ob: &PyAny) -> PyResult<DateTime<Utc>> {
+        let dt = <PyDateTime as PyTryFrom>::try_from(ob)?;
+        let ms = dt.get_fold() as u32 * 1_000_000 + dt.get_microsecond();
+        let h = dt.get_hour().into();
+        let m = dt.get_minute().into();
+        let s = dt.get_second().into();
+        let tz = if let Some(tzinfo) = dt.get_tzinfo() {
+            tzinfo.extract()?
+        } else {
+            return Err(PyTypeError::new_err("Not datetime.timezone.utc"));
+        };
+        let dt = NaiveDateTime::new(
+            NaiveDate::from_ymd(dt.get_year(), dt.get_month().into(), dt.get_day().into()),
+            NaiveTime::from_hms_micro(h, m, s, ms),
+        );
+        Ok(DateTime::from_utc(dt, tz))
+    }
+}
+
+impl ToPyObject for FixedOffset {
+    fn to_object(&self, py: Python<'_>) -> PyObject {
+        let dt_module = py.import("datetime").expect("Failed to import datetime");
+        let dt_timezone = dt_module
+            .getattr("timezone")
+            .expect("Failed to getattr timezone");
+        let seconds_offset = self.local_minus_utc();
+        let td =
+            PyDelta::new(py, 0, seconds_offset, 0, true).expect("Failed to contruct timedelta");
+        let offset = dt_timezone
+            .call1((td,))
+            .expect("Failed to call timezone with timedelta");
+        offset.into()
+    }
+}
+
+impl IntoPy<PyObject> for FixedOffset {
+    fn into_py(self, py: Python<'_>) -> PyObject {
+        ToPyObject::to_object(&self, py)
+    }
+}
+
+impl FromPyObject<'_> for FixedOffset {
+    /// Convert python tzinfo to rust [`FixedOffset`].
+    ///
+    /// Note that the conversion will result in precision lost in microseconds as chrono offset
+    /// does not supports microseconds.
```

Comment:
@djc I guess even with chrono changes to `TimeDelta` this will stay the same?

Reply:
The `TimeDelta` changes we're thinking about for 0.5 don't have much to do with the `FixedOffset` you're using here, as far as I know. We could consider increasing precision in `FixedOffset::local_minus_utc` if you think that's useful? That might conceivably be done in a backwards compatible version for 0.4, too.

Reply:
That would be good I guess if we increase precision in microseconds, I think it could have some impact to users as it it needs to store another byte in some cases but that would mean that there are no precision loss in conversion between chrono FixedOffset and python timedelta.